calico_containers/integrations/kubernetes/calico_kubernetes.py

```diff
@@ -95,19 +96,19 @@ def _configure_interface(self):
         return ep
 
     def _get_node_ip(self):
-        """Determine the IP for the host node.
-
-        This hits a well-known IP (the Google public DNS).
+        """ 
+            Determine the IP for the host node.
         """
-        # We'd like to use the k8s API here, but it doesn't work in the Vagrant
-        # case since the node's name is set to its IP (so we have no way of
-        # getting from hostname=>IP).
-        # TODO: do this more reliably by parsing 'ip
-        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-        s.connect(('8.8.8.8', 80))
-        ip = s.getsockname()[0]
-        s.close()
-        return ip
+        # Compile list of IPv4 addresses on network, return the first entry
+        ipv4_addresses = get_host_ips(version=4)
+        
+        # If no IPv4 addresses exist, check IPv6, return the first entry
+        if not ipv4_addresses:
+            return get_host_ips(version=6)[0]
```

Comment:
There's no guarantee that get_host_ips(version=6) will return any addresses here, so just blindly grabbing index 0 might raise an IndexError.

Might want to try something like this:

```
# Get the IPs for this node.  Try getting IPv4 addresses, if there are none use IPv6 addresses.
addrs = get_host_ips(version=4) or get_host_ips(version=6)

if not addrs:
    # No configured IP addresses - return None.
    return None

# An address exists.
return addrs[0]
```

Another options might be 
```
try:
    # Try to get an address
    return addrs[0]
except IndexError:
    # No addresses found.
    return None
```